TaskLayer/XLSearchTask/PostXLSearchAnalysisTask.cs

```diff
@@ -136,33 +114,51 @@ public MyTaskResults Run(string OutputFolder, List<DbForTask> dbFilenameList, Li
             return MyTaskResults;
         }
 
+        public void ComputeXlinkQandPValues(List<CrosslinkSpectralMatch> allPsms, List<CrosslinkSpectralMatch> intraCsms, List<CrosslinkSpectralMatch> interCsms, CommonParameters commonParameters, string taskId)
+        {
+            // calculate FDR
+            DoCrosslinkFdrAnalysis(interCsms);
+            DoCrosslinkFdrAnalysis(intraCsms);
+            
+            List<CrosslinkSpectralMatch> crossCsms = allPsms.Where(p => p.CrossType == PsmCrossType.Inter || p.CrossType == PsmCrossType.Intra).ToList();
+            new FdrAnalysisEngine(crossCsms.ToList<PeptideSpectralMatch>(), 0, commonParameters, new List<string>() { taskId }, "crosslink").Run();
+
+            List<CrosslinkSpectralMatch> singles = allPsms.Where(p => p.CrossType != PsmCrossType.Inter).Where(p => p.CrossType != PsmCrossType.Intra).ToList();
```

Comment:
Change to: Where(p => p.CrossType != PsmCrossType.Inter && p.CrossType != PsmCrossType.Intra)

Reply:
The reason that I do this is so that I can use separate criteria that are a better match for the peptide classes. For single, deadend, looplink, I can use retention times, etc. For the xlinks, I can use all the beta peptide info and not the other stuff. I think that the PEP is more accurate this way.